osu.Game.Tests/Beatmaps/WorkingBeatmapTest.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using NUnit.Framework;
+using osu.Game.Rulesets.Osu;
+using osu.Game.Rulesets.Osu.Mods;
+
+namespace osu.Game.Tests.Beatmaps
+{
+    [TestFixture]
+    public class WorkingBeatmapTest
+    {
+        [Test]
+        public void TestModsApplicableToDifficultyReadFromDifficulty()
+        {
+            var beatmap = new TestBeatmap(new OsuRuleset().RulesetInfo)
+            {
+                BeatmapInfo =
+                {
+                    BaseDifficulty =
+                    {
+                        OverallDifficulty = 11,
+                        ApproachRate = 11,
+                        DrainRate = 11,
+                        CircleSize = 11
+                    }
+                }
+            };
+
+            var workingBeatmap = new TestWorkingBeatmap(beatmap);
+
+            var playableBeatmap = workingBeatmap.GetPlayableBeatmap(new OsuRuleset().RulesetInfo, new[]
+            {
+                new OsuModDifficultyAdjust()
+            });
+
+            // Playable beatmap should have read default difficulty settings from beatmap.
+            Assert.That(playableBeatmap.BeatmapInfo.BaseDifficulty.OverallDifficulty, Is.EqualTo(11));
+            Assert.That(playableBeatmap.BeatmapInfo.BaseDifficulty.ApproachRate, Is.EqualTo(11));
+            Assert.That(playableBeatmap.BeatmapInfo.BaseDifficulty.DrainRate, Is.EqualTo(11));
+            Assert.That(playableBeatmap.BeatmapInfo.BaseDifficulty.CircleSize, Is.EqualTo(11));
```

Comment:
I think the mod values were  supposed to be checked here, right? It's the mod that's getting `ReadFromDifficulty()` called on in this commit.

Reply:
Whether the mod is checked or the beatmap is checked is the same thing in the end.
There are about 3 different ways this could be asserted - check the resultant beatmap (as I’ve done), check the mod, or check that ReadFromDifficulty is called with a custom override.

Reply:
After reverting the fix the test still passes, though. Unless I got something royally wrong but I'm pretty sure I checked a few times.

Reply:
On second thought I was testing with the first commit removed, so that may have been running interference...

Reply:
First commit should be unrelated. I definitely tested this though… if it doesn’t work I’ll have to look into it.

Reply:
Actually I think I only tested this with the changes in my previous PR, which fixed DA not always applying… welp